Source/objects.cpp

```diff
@@ -1164,140 +1164,130 @@ void AddShrine(int i)
 		val = GenerateRnd(shrines);
 	} while (!slist[val]);
 
-	Objects[i]._oVar1 = val;
+	shrine._oVar1 = val;
 	if (!FlipCoin()) {
-		Objects[i]._oAnimFrame = 12;
-		Objects[i]._oAnimLen = 22;
+		shrine._oAnimFrame = 12;
+		shrine._oAnimLen = 22;
 	}
 }
 
-void AddBookcase(int i)
+void AddBookcase(Object &bookcase)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
-	Objects[i]._oPreFlag = true;
+	bookcase._oRndSeed = AdvanceRndSeed();
+	bookcase._oPreFlag = true;
 }
 
-void AddBookstand(int i)
+void AddBookstand(Object &bookstand)

```

Comment:
The other big cleanup that would be worth including from #4571 is combining all these functions which only set `Object::_oRndSeed` :D